sklearn/calibration.py

```diff
@@ -322,6 +339,16 @@ def fit(self, X, y, sample_weight=None):
                     "incorrect."
                 )
 
+            if self.class_weight is not None:
+                # Build sample weights for calibrator.
+                # Those weights will not be used for the training of
+                # the underlying estimator which will then be calibrated.
+                self.class_weight_ = compute_class_weight(
+                    self.class_weight, classes=self.classes_, y=y
+                )
+                label_encoder_ = LabelEncoder()
+                sample_weight *= self.class_weight_[label_encoder_.fit_transform(y)]
```

Comment:
I might be confused but does this not mean that estimator will trained with these weights in `_fit_classifier_calibrator_pair` here:

https://github.com/scikit-learn/scikit-learn/blob/b84d67751753cd7ee81b3ede8d3d16951ddc4cef/sklearn/calibration.py#L530-L532

?

Reply:
Thank you for your comment @lucyleeow! I modified the code so that the estimator and the calibrator are trained with different sample weights.

Reply:
Thanks @amascia - I do have a question, what is the theory behind using sample weights for the calibrator but not the estimator?